### ISSUE TITLE
Many useless rules and group-bundle is not override

### ISSUE THREAD

**Issue by nolouch** [pingcap/tidb#28353]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

use master creates a new tidb cluster.

### 2. What did you expect to see? (Required)
only one rule:
```
[
  {
    "group_id": "pd",
    "id": "default",
    "start_key": "",
    "end_key": "",
    "role": "voter",
    "count": 3
  }
]
```

### 3. What did you see instead (Required)

many rules:
<details>
<summary> detail rules </summary>

```
[
  {
    "group_id": "TiDB_DDL_11",
    "id": "voter",
    "start_key": "7480000000000000ff0b00000000000000f8",
    "end_key": "7480000000000000ff0c00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "TiDB_DDL_13",
    "id": "voter",
    "start_key": "7480000000000000ff0d00000000000000f8",
    "end_key": "7480000000000000ff0e00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "TiDB_DDL_15",
    "id": "voter",
    "start_key": "7480000000000000ff0f00000000000000f8",
    "end_key": "7480000000000000ff1000000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "TiDB_DDL_17",
    "id": "voter",
    "start_key": "7480000000000000ff1100000000000000f8",
    "end_key": "7480000000000000ff1200000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "TiDB_DDL_19",
    "id": "voter",
    "start_key": "7480000000000000ff1300000000000000f8",
    "end_key": "7480000000000000ff1400000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_21",
    "id": "voter",
    "start_key": "7480000000000000ff1500000000000000f8",
    "end_key": "7480000000000000ff1600000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_23",
    "id": "voter",
    "start_key": "7480000000000000ff1700000000000000f8",
    "end_key": "7480000000000000ff1800000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_25",
    "id": "voter",
    "start_key": "7480000000000000ff1900000000000000f8",
    "end_key": "7480000000000000ff1a00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_27",
    "id": "voter",
    "start_key": "7480000000000000ff1b00000000000000f8",
    "end_key": "7480000000000000ff1c00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_29",
    "id": "voter",
    "start_key": "7480000000000000ff1d00000000000000f8",
    "end_key": "7480000000000000ff1e00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_31",
    "id": "voter",
    "start_key": "7480000000000000ff1f00000000000000f8",
    "end_key": "7480000000000000ff2000000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_33",
    "id": "voter",
    "start_key": "7480000000000000ff2100000000000000f8",
    "end_key": "7480000000000000ff2200000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_35",
    "id": "voter",
    "start_key": "7480000000000000ff2300000000000000f8",
    "end_key": "7480000000000000ff2400000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_37",
    "id": "voter",
    "start_key": "7480000000000000ff2500000000000000f8",
    "end_key": "7480000000000000ff2600000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_39",
    "id": "voter",
    "start_key": "7480000000000000ff2700000000000000f8",
    "end_key": "7480000000000000ff2800000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_41",
    "id": "voter",
    "start_key": "7480000000000000ff2900000000000000f8",
    "end_key": "7480000000000000ff2a00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_43",
    "id": "voter",
    "start_key": "7480000000000000ff2b00000000000000f8",
    "end_key": "7480000000000000ff2c00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_45",
    "id": "voter",
    "start_key": "7480000000000000ff2d00000000000000f8",
    "end_key": "7480000000000000ff2e00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638956
  },
  {
    "group_id": "TiDB_DDL_47",
    "id": "voter",
    "start_key": "7480000000000000ff2f00000000000000f8",
    "end_key": "7480000000000000ff3000000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638957
  },
  {
    "group_id": "TiDB_DDL_49",
    "id": "voter",
    "start_key": "7480000000000000ff3100000000000000f8",
    "end_key": "7480000000000000ff3200000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638957
  },
  {
    "group_id": "TiDB_DDL_5",
    "id": "voter",
    "start_key": "7480000000000000ff0500000000000000f8",
    "end_key": "7480000000000000ff0600000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "TiDB_DDL_51",
    "id": "voter",
    "start_key": "7480000000000000ff3300000000000000f8",
    "end_key": "7480000000000000ff3400000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638957
  },
  {
    "group_id": "TiDB_DDL_53",
    "id": "voter",
    "start_key": "7480000000000000ff3500000000000000f8",
    "end_key": "7480000000000000ff3600000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638957
  },
  {
    "group_id": "TiDB_DDL_7",
    "id": "voter",
    "start_key": "7480000000000000ff0700000000000000f8",
    "end_key": "7480000000000000ff0800000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "TiDB_DDL_9",
    "id": "voter",
    "start_key": "7480000000000000ff0900000000000000f8",
    "end_key": "7480000000000000ff0a00000000000000f8",
    "role": "voter",
    "count": 3,
    "label_constraints": [
      {
        "key": "engine",
        "op": "notIn",
        "values": [
          "tiflash"
        ]
      }
    ],
    "create_timestamp": 1632638955
  },
  {
    "group_id": "pd",
    "id": "default",
    "start_key": "",
    "end_key": "",
    "role": "voter",
    "count": 3
  }
]
```

</details>

it will create a rule for each table, and I think the rule is useless. and each table's rule is not overridden. which causes **6 replicas(table rule + pd default rule = 6 ) for each table** that doesn't meet expectations.
### 4. What is your TiDB version? (Required)

master, after https://github.com/pingcap/tidb/pull/28037

**Comment by nolouch** [comment]
cc @AilinKid  @xhebox

**Comment by nolouch** [comment]
ref https://github.com/pingcap/tidb/issues/28364

**Comment by nolouch** [comment]
BTW, I think we should reduce the meaningless rules. Too many rules will cause the search space of PD rules to become larger. And too many rules stored in etcd may cause some performance problems. @AilinKid

**Comment by xhebox** [comment]
> BTW, I think we should reduce the meaningless rules. Too many rules will cause the search space of PD rules to become larger. And too many rules stored in etcd may cause some performance problems. @AilinKid

It will also be fixed in #28365. So I am going to closed this issue in favor of #28364.
